### PR TITLE
fix: sft training, reproduce openpi curves

### PR DESCRIPTION
**Pull request by red0orange**
fix: resolve optimizer config and fsdp gradient sync bugs

This PR aligns the training configuration with the official OpenPI setup to fully reproduce the training curves.

### Description
1. **Enable full-parameter fine-tuning**: Changed `train_expert_only` to `False` to match the official training strategy.
2. **Align optimizer parameters**: Replicated the exact optimizer configuration and strategy from the official OpenPI setup.
3. **Fix FSDP synchronization bug**: Identified and resolved a critical bug in `fsdp_sft_worker` where automatic gradient synchronization (all-reduce) was failing during multi-GPU training. Although the code appeared to block and wait, the gradients were not actually syncing, leading to inconsistent `grad_norm` across ranks. I added a manual full-parameter gradient synchronization step after `loss.backward()` to fix this.

### Motivation and Context
This PR addresses the unexpected performance degradation reported in Issues #540.
Previous attempts to fine-tune Pi0 resulted in regression despite normal-looking logs. After investigation (with significant help from @XuS1994), we identified that the regression was caused by a combination of partial fine-tuning (expert-only) and a silent failure in gradient synchronization during multi-GPU training.

These changes ensure that:
- The training strategy matches the official implementation (full-parameter).
- Gradients are correctly synchronized across all GPUs, preventing divergent model updates.

### How has this been tested?
- **Single-GPU verification**: Confirmed that changes 1 and 2 (config and full-param tuning) successfully reproduce the OpenPI training curves on a single GPU.
- **Multi-GPU verification**: Validated the FSDP sync fix by monitoring `grad_norm` values across different ranks. Before the fix, `grad_norm` values varied across cards; after the fix, they are consistent, and the training loss decreases as expected.
- **End-to-end training**: Completed a full SFT run to verify that the performance regression is resolved.

### Additional information
Credits: Thanks to @XuS1994 for the assistance.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by zoeyuchao** [comment]
@red0orange Hi thanks for your contribution！The lint and pr title need to be modified a little bit, please see the ci test

**Comment by red0orange** [comment]
> @red0orange Hi thanks for your contribution！The lint and pr title need to be modified a little bit, please see the ci test

Thanks for the reminder, I'll fix them.

File: examples/sft/config/libero_sft_openpi.yaml [hunk 1]
```diff
@@ -23,10 +23,10 @@ runner:
     experiment_name: "test_openpi"
     logger_backends: ["tensorboard"] # wandb, swanlab
 
-  max_epochs: 1000
+  max_epochs: 30000
   max_steps: -1
   val_check_interval: -1
-  save_interval: 10
+  save_interval: 2000
```
Comment (LiuZhihao2022) [inline examples/sft/config/libero_sft_openpi.yaml:29]: Maybe too frequent. 15 ckpt will be saved in 30000 steps

File: rlinf/workers/sft/fsdp_sft_worker.py [hunk 2]
```diff
@@ -138,10 +139,29 @@ def run_training(self):
                         )
                     loss = losses.mean()
 
+                total_loss += loss.item()
                 loss = loss / self.gradient_accumulation
                 with backward_ctx:
                     self.grad_scaler.scale(loss).backward()
 
+            # Manual gradient sync (temporary fix)
```
Comment (LiuZhihao2022) [inline rlinf/workers/sft/fsdp_sft_worker.py:147]: It seems that parameters are averaged per group. Does this also hold under FSDP’s full_shard mode?

Comment (red0orange) [reply rlinf/workers/sft/fsdp_sft_worker.py:147]: Actually I missed that. I only tested it on NO_SHARD—can we assume SFT usually requires NO_SHARD anyway? I can add a check to ensure this only runs if sharding_strategy is NO_SHARD.

Comment (LiuZhihao2022) [reply rlinf/workers/sft/fsdp_sft_worker.py:147]: In our experiments, using FULL_SHARD does indeed significantly reduce GPU memory usage. Therefore, if it’s convenient for you, it would be appreciated if you could add the corresponding check to ensure FULL_SHARD also works well😊.

**Comment by zoeyuchao** [comment]
@red0orange Thank you for your hard work! We solved this problem at https://github.com/RLinf/RLinf/pull/568, it has merged your code, so we closed this pr. Thanks again!

**Comment by red0orange** [comment]
> @red0orange Thank you for your hard work! We solved this problem at #568, it has merged your code, so we closed this pr. Thanks again!

I’m glad to contribute code to RLinf — I think the corresponding issue #540 can be closed now as well.
